odoo/addons/base/models/res_users.py

```diff
@@ -189,6 +189,14 @@ def SELF_WRITEABLE_FIELDS(self):
         """
         return ['signature', 'action_id', 'company_id', 'email', 'name', 'image_1920', 'lang', 'tz', 'api_key_ids', 'phone']
 
+    @api.model
+    @tools.ormcache()
+    def _self_accessible_fields(self) -> tuple[frozenset[str], frozenset[str]]:
+        """Readable and writable fields by portal users."""
+        readable = frozenset(self.SELF_READABLE_FIELDS)
+        writeable = frozenset(self.SELF_WRITEABLE_FIELDS)
+        return readable, writeable
```

Comment:
Can you make it deterministic?

Reply:
Added `sorted` in 2 uses that need the order as it is simpler than putting frozendicts in the cache.